src/RetryTrait.php

```diff
@@ -58,7 +59,7 @@ private function checkShouldRetryAgain($retryAttempt)
                 $retryAttempts
             );
         } elseif ($retryFor = $this->getRetryForSecondsAnnotation()) {
-            if (is_null(self::$timeOfFirstRetry)) {
+            if (self::$timeOfFirstRetry === null) {
```

Comment:
Why are you preferring `=== null` to `is_null`?

Reply:
Unless you are dealing with loose types, i.e. if you were using `==`, `is_null()` brings no benefit (picking one or the other is a stylistic choice) but comparing against null directly is relying on a native construct rather than a function, which makes it significantly faster. That said it remains within the realms of micro-optimization: it's just that no form is superior to the other so you might just pick the most performant one